tests/components/solarlog/test_config_flow.py

```diff
@@ -125,40 +125,27 @@ async def test_form_exceptions(
 
 async def test_abort_if_already_setup(hass: HomeAssistant, test_connect: None) -> None:
     """Test we abort if the device is already setup."""
-    flow = init_config_flow(hass)
+
     MockConfigEntry(
         domain="solarlog", data={CONF_NAME: NAME, CONF_HOST: HOST}
     ).add_to_hass(hass)
 
-    # Should fail, same HOST different NAME (default)
-    result = await flow.async_step_user(
-        {CONF_HOST: HOST, CONF_NAME: "solarlog_test_7_8_9", "extended_data": False}
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
-    assert result["type"] is FlowResultType.FORM
-    assert result["errors"] == {CONF_HOST: "already_configured"}
+    await hass.async_block_till_done()
 
-    # Should fail, same HOST and NAME
-    result = await flow.async_step_user({CONF_HOST: HOST, CONF_NAME: NAME})
     assert result["type"] is FlowResultType.FORM
-    assert result["errors"] == {CONF_HOST: "already_configured"}
-
-    # SHOULD pass, diff HOST (without http://), different NAME
-    result = await flow.async_step_user(
-        {CONF_HOST: "2.2.2.2", CONF_NAME: "solarlog_test_7_8_9", "extended_data": False}
-    )
-    assert result["type"] is FlowResultType.CREATE_ENTRY
-    assert result["title"] == "solarlog_test_7_8_9"
-    assert result["data"][CONF_HOST] == "http://2.2.2.2"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
 
-    # SHOULD pass, diff HOST, same NAME
-    result = await flow.async_step_user(
-        {CONF_HOST: "http://2.2.2.2", CONF_NAME: NAME, "extended_data": False}
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: HOST, CONF_NAME: "solarlog_test_7_8_9", "extended_data": False},
     )
     await hass.async_block_till_done()
```

Comment:
Can also be removed

Reply:
You mean `await hass.async_block_till_done()`, right?